File: java-api-core/test/com/kinvey/java/model/AggregationTest.java
package com.kinvey.java.model;

import com.kinvey.java.core.KinveyMockUnitTest;

import java.util.ArrayList;

/**
 * Created by edward on 7/31/15.
 */
public class AggregationTest extends KinveyMockUnitTest {

    private Aggregation agg;

    public void testConstruction(){
        agg = new Aggregation(new ArrayList<Aggregation.Result>());
        assertNotNull(agg.getResults());
        agg = new Aggregation(null);
        assertNotNull(agg.getResults());
        Aggregation.Result res = new Aggregation.Result();
        assertNull(res.getResult());
    }

    public void testResults(){
        Aggregation.Result res = new Aggregation.Result();
        res.setResult(1);
        res.put("key", "value");

        ArrayList<Aggregation.Result> list = new ArrayList<Aggregation.Result>();
        list.add(res);
        agg = new Aggregation(list);
        assertEquals(1, agg.getResultsFor("key", "value").get(0));
        assertEquals(0, agg.getResultsFor("key", "novalue").size());
        assertEquals(0, agg.getResultsFor("nokey", "novalue").size());
    }

}

File: java-api-core/test/com/kinvey/java/model/AggregationTest.kt
package com.kinvey.java.model

import com.kinvey.java.core.KinveyMockUnitTest
import com.kinvey.java.dto.BaseUser
import com.kinvey.java.model.Aggregation.Result
import java.util.*

/**
 * Created by edward on 7/31/15.
 */
class AggregationTest : KinveyMockUnitTest<BaseUser>() {

    private var agg: Aggregation? = null

    fun testConstruction() {
        agg = Aggregation(ArrayList())
        assertNotNull(agg?.results)
        agg = Aggregation(null)
        assertNotNull(agg?.results)
        val res = Result()
        assertNull(res.result)
    }

    fun testResults() {
        val res = Result()
        res.result = 1
        res["key"] = "value"
        val list = listOf(res)
        agg = Aggregation(list)
        assertEquals(1, agg?.getResultsFor("key", "value")!![0])
        assertEquals(0, agg?.getResultsFor("key", "novalue")?.size)
        assertEquals(0, agg?.getResultsFor("nokey", "novalue")?.size)
    }
}